exonum/src/events/noise/wrappers/sodium_wrapper/handshake.rs

```diff
@@ -44,20 +51,57 @@ impl HandshakeParams {
         secret_key: SecretKey,
         connect_list: SharedConnectList,
         max_message_len: u32,
+        address: SocketAddr,
     ) -> Self {
-        let (public_key, secret_key) = into_x25519_keypair(public_key, secret_key).unwrap();
+        let (public_key_x25519, secret_key_x25519) =
+            into_x25519_keypair(public_key, secret_key.clone()).unwrap();
 
         HandshakeParams {
-            public_key,
-            secret_key,
+            public_key_x25519,
+            secret_key_x25519,
             max_message_len,
             remote_key: None,
+            remote_key_x25519: None,
             connect_list,
+            connect_info: ConnectInfo {
+                address,
+                public_key,
+            },
+            public_key,
+            secret_key,
         }
     }
 
     pub fn set_remote_key(&mut self, remote_key: PublicKey) {
-        self.remote_key = Some(into_x25519_public_key(remote_key));
+        self.remote_key = Some(remote_key);
+        self.remote_key_x25519 = Some(into_x25519_public_key(remote_key));
+    }
+
+    pub fn peer_list(&self) -> Option<PeerList> {
```

Comment:
See the comment about the struct fields. The very existence of this method tells me that `HandshakeParams` is designed incorrectly.